Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionActionCocoa.mm

```diff
@@ -532,7 +633,96 @@ - (void)_dismissPopup
 
     return m_popupViewController.get();
 }
-#endif
+#endif // PLATFORM(IOS_FAMILY)
+
+#if PLATFORM(MAC)
+NSPopover *WebExtensionAction::popupPopover()
+{
+    if (!presentsPopup())
+        return nil;
+
+    if (m_popupPopover)
+        return m_popupPopover.get();
+
+    m_popupPopover = [[_WKWebExtensionActionPopover alloc] initWithWebExtensionAction:*this];
+
+    return m_popupPopover.get();
+}
+
+void WebExtensionAction::setPopupPopoverAppearance(Appearance appearance)
+{
+    m_popoverAppearance = appearance;
```

Comment:
any reason on why we want to set the appearance before checking if there's a popup?